### ISSUE TITLE
[Feature request] put search results in order

### ISSUE THREAD

**Issue by katerynakocherhina** [osmandapp/Osmand#8613]
When searching for a certain city, it would be useful if there was an opportunity to see the results put in order from the closest city to the furthest one from the user's current location.

**Comment by dmpr0** [comment]
## ToDo
The goal of this task is to introduce sorting capabilities for search results and refine the search result screen's layout. This involves migrating the **Show on map** action to a Floating Action Button (FAB) at the bottom and modernizing the top control bar using Material Design Chips for the **Sort** and **Filter** actions.

<table style="width:100%">
  <tr>
    <th>Filter bar</th>
    <th>Sort by menu</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/f0727b09-2b41-4690-bb4e-3c05169b53dd"  width="200" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/3b3a10a0-2543-4821-9441-cfc4c4fe9ce3"  width="200" height="auto"/> </td>
  </tr>
</table>   

## Top Bar Redesign (Filters & Sort)
<table style="width:100%">
  <tr>
    <th>Bar</th>
    <th>Menu</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/c0753b5a-94b6-43c4-90a4-5033b042549b"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/012b3d5f-75ed-414b-8e3c-700ebf7a36bd"  width="300" height="auto"/> </td>
  </tr>
</table>   

- [x] Remove the background from the top bar to make it cleaner.
- [x] Remove the inline **Show on map** button from the top (it will be moved to a FAB).
- [x] Implement Material Design [Filter Chips](https://developer.android.com/develop/ui/compose/components/chip) for the new interactive buttons.

### Sort chip
- [x] Add a new "Sort by" Chip.
- [x] Add leading icon – `ic_action_sort`
- - [x] Add a trailing downward arrow icon on the right side of the chip.
- - [x] On tap: Open a bottom menu/dropdown to select the sorting method.
- [x] **Relevance:** Sort the search result list by relevance. 
- - [x] Set this as the default state  on all categories except Nearest POI.
- [x] **Nearest:** Sort the search result list purely by distance (closest to the user's current location or the map's current center point).

### Filter chip

<table style="width:100%">
  <tr>
    <th>Default</th>
    <th>Focused</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/9635f153-8f58-427f-aa70-7f0f8ca01350"  width="auto" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/9c519cd5-ec8d-48e4-bb88-4823cef856b4"  width="auto" height="auto"/> </td>
  </tr>
</table>   

- [x] Replace the existing standalone filter icon with a standard Chip.
- [x] Place as the first item in the filter bar
- [x] Use icon: `ic_action_filter`
- [x] When the user selects one or more filters, add the quantity of selected filters next to the icon: `3`
- [ ] Default state background color –
- [ ] Focused state – 

### Appearance

**Default state:**
- [x] Background color – `chip_bg_light/dark`
- [x] Outline color – `btn_outline_secondary_light/dark`

**Focused state:**
- [x] Background color – `btn_bg_accent_light/dark`
- [x] Outline color – `btn_outline_accent_light/dark`


### Behavior 
Apply only to second row of filters.

- [x] Move used filter to to the first place. For example before selection "Diet; Vegan; Takeaways", after selecting Takeaway, "Takeaway; Diet; Vegan"

## "Show on map" FAB

<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/74db2973-d659-45b5-a617-5973b77c3e69"  width="300" height="auto"/> </td>
  </tr>
</table>   

- [x] Add an Extended Floating Action Button (FAB) anchored to the bottom of the screen.
- [x] Without text, use icon – `ic_show_on_map_outlined`
- [x] Tap action: Close the list and display the current search results on the map.

## Search around

<table style="width:100%">
  <tr>
    <th>New menu</th>
    <th>Current implementation</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/9f88c5a0-649c-4d05-9b8c-58d74a2f87cd"  width="300" height="auto"/> </td>  
 <td><img src="https://github.com/user-attachments/assets/740dacab-2643-4345-9574-8501cacede82"  width="300" height="auto"/> </td>
  </tr>
</table>   

### "Search around" Dropdown Chip
- [x] Add a new dropdown chip to control the search area context.
- [x] **Visibility Condition:** Display this chip *only* when the search is opened significantly away from the user's current location (reuse the existing logic/threshold currently used to trigger the "Search 88.2 km away" banner).

**Dropdown Options:**

**1. Map center (Set by Default)**
- [x] Icon: `ic_show_on_map_outlined`
- [x] Description: Display the dynamic distance to the map center (e.g., `556 km away from your location`). Ensure the distance format uses proper units and calculations.

**2. My location**
- [x] Icon: `ic_action_location_marker_outlined`
- [x] Description: `Search around your position`
- [x] Behavior: Tapping this option should immediately perform the search around the user's current GPS location (this should function exactly identically to the current dedicated "location" button).


## Categories and Filters
We will show list of categories for founded results, for search by name and top filters for search by category


### By category
When searching by selecting category

- [x] Show filters with `top="true"`

### By Name

<table style="width:100%">
  <tr>
    <th></th>
    <th></th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/1884f333-041f-40a3-be49-ea503f3d881c"  width="200" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/975a1965-044c-4448-91e1-dc528011dba3"  width="200" height="auto"/> </td>
  </tr>
</table>   


- [x] Display a horizontally scrollable list of category chips for the found search results. 
- [x] Show categories based on the current search output
- [x] **Interaction & Reordering:**
- - [x] Tapping a category chip filters the list to show only results matching that specific category. 
- - [x] **Move selected to front:** Once a category is selected, move it to the first position in the horizontal list
- - [x] Tapping a selected chip again should deselect it and return the list to the full search results.
- - [x] Allow to select multiply categories 

## Opening hours

Separate task

**Comment by Korb** [comment]
If I understand correctly, the lack of sorting search results in order of increasing distance to them is currently absent from the OsmAnd:

<details markdown="1">
<summary>Click to see screenshot</summary>

---

<img width="1080" height="1507" alt="Image" src="https://github.com/user-attachments/assets/76ef84ff-e935-4573-b880-72827e7d8313" />

--- 

</details>

However, in the future, when this feature is added, the question will almost certainly arise: what kind of distance is meant? Linear distance along the shortest route (aircraft navigation) or taking into account the route length on the selected means of transportation? In rural areas, the closest object in direct distance is not always closer in terms of travel distance or time to reach it.

--- 

OsmAnd 5.3.10
Android 14 (Funtouch OS 14)
Device: Vivo T1 (4G) / V2153

**Comment by dmpr0** [comment]
## Additional

<table style="width:100%">
  <tr>
    <th>Current</th>
    <th>Updated</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/c874bf01-0e34-4ee4-816e-c2455fdc215c"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/0f102b4f-df6f-4113-95bc-dd10a8b0b347"  width="300" height="auto"/> </td>
  </tr>
</table>   

Replace the **Advanced coordinates search** text button below the search bar with a separate item above the search results.

- [x] Show the **Advanced coordinates search** item using the same logic as used for the text button
- [x] Use `ic_action_coordinates_location`
- [x] On tap, open Coordinates search as it was before

**Comment by dmpr0** [comment]
## UI Review



- [x] Set icon size to `18x18` dp in Chip

- - - - - -
 

<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/322875c5-2838-4dfe-9867-47fb5f3d1c64"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Only categories from selected category should be visible on the second line

- - - - - -
 
<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/322875c5-2838-4dfe-9867-47fb5f3d1c64"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  


- [x] FAB is under Navigation bar

- - - - - -
 
<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/e52e8c19-e4d2-4ca2-ad3b-53c7eda485c7"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Paddings (left/right) should not be visible while scroll

- - - - - -

<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/50f19a61-8018-4ffe-ac5a-2233ded161c7"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Dark theme 

- - - - - -

<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/59d9c1d7-1cb9-47aa-a820-53df4f611743"  width="300" height="auto"/> </td>
        <td><img src="https://github.com/user-attachments/assets/4974ccd7-51d1-4fe4-9e77-8bb308327a82"  width="300" height="auto"/> </td>
        <td><img src="https://github.com/user-attachments/assets/c0708206-6c90-4f65-b486-a2669df17176"  width="300" height="auto"/> </td>
      </tr>
    </table>   
</details>   
 

- [x] Dropdown menu looks to big, align it sizes with default ones

**Comment by Zirochkabila** [comment]
# Testing 1

> OsmAnd~ 5.4.0#8138m, released: 2026-07-02

**Severity**: 🟠 Major / **Priority**: 🟠 Normal

* [x] 1. Filters selected for a category remain active throughout the entire session and are cleared only after OsmAnd is reopened.

<details><summary>Details</summary>
<video src="https://github.com/user-attachments/assets/b4506c43-6bd0-4636-be40-f05d8c9d4e9a"></video>
</details>

**Severity**: 🟡 Minor / **Priority**: 🟠 Normal

* [x] 1. The filter icon remains visible for a few more seconds after leaving the category.

<details><summary>Details</summary>
<video src="https://github.com/user-attachments/assets/a8492387-4bc4-45e2-8547-e69db41a7d6c"></video>
</details>

* [x] 2. After switching to Light Theme, the categories are displayed in a dark color until the app is restarted.

<details><summary>Details</summary>
<img width="400" alt="Screenshot 2023-01-31 at 11 19 41" src="https://github.com/user-attachments/assets/0517f26c-b72a-4d24-8075-00c13ed2c3d3">
</details>

**Comment by vshcherb** [comment]
**Urgent**
- [x] Chips should represent poi types from the actual search! Filter should be applied instant without research
- [ ] "Open now" chip should be added first
- [x] Chips should not change position on click
- [x] Chips should be sorted by popularity in represented search
- [x] Chips should be poi types and not Poi category!

**Comment by Zirochkabila** [comment]
- [x] If the navigation system is set to “Three Buttons” instead of “Gestures,” the “Show on Map” and “Save” buttons overlap each other

<details><summary>Details</summary>
<img width="400" alt="Screenshot 2023-01-31 at 11 19 41" src="https://github.com/user-attachments/assets/6dfc85e2-d512-4bc7-84f4-1f6df15fc827">
</details>

**Comment by isso-git** [comment]
Thanks for working on the search, I try it a little bit and have two points I would mention.

If I select a poi type Chip and after this change the sort order from "Nearest" to "Relevance", than the previous selection of the poi type is gone.
I think, in this case the selection of the poi type should stay active.
But I have no idea how that can be combined with the proposed sorting of the poi type by popularity in represented search.

I would like to see a extra dropdown menu for the selection of the many poi types chips, like it is for folder selection in adding favorites menu.
But maybe limiting the search as planned, to poi types only (without categories), will solve the problem, given the smaller number of results.

Thanks for listening.
Tested with nightly 4.5.0#8242 (2026-07-13)

**Comment by Zirochkabila** [comment]
# Testing 2

> OsmAnd~ 5.4.0#8288m, released: 2026-07-17

**Severity**: 🟠 Major / **Priority**: 🟠 Normal

* [x] 1. The **Show on Map** button used to disappear, and on Android 15 the **Save** button appears in the row with the navigation buttons.

<details><summary>Details</summary>
<table style="width:100%">
  <tr>
    <th>Android 15</th>
    <th>Android 12</th>
  </tr>
  <tr>
    <th><img src="https://github.com/user-attachments/assets/a66bc521-f007-468f-a59a-268f65dcf6fb" width="300" height="auto"/></th>
    <th><img src="https://github.com/user-attachments/assets/44c35eae-1f01-4c5c-8f70-c48f38e6f2a1" width="300" height="auto"/></th>
  </tr>
</table>
</details>

**Comment by dmpr0** [comment]
## UI Review

<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/34c83f3f-ec14-42f7-b37a-68be6563cd25"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Bottom part of the chips in both first and second lines are cropped. Left / right stroke is 3 px, bottom is only 2px
- [x] What color is used for grey stroke for chips?
- [x] Use `icon_color_primary_` for grey icons in chips

**Selected chip**

<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/e7732425-d77b-4adc-aa36-6b8c9dd0ac04"  width="300" height="auto"/> </td>
  </tr>
</table>   


- [x] Show checkmark as leading icon for selected chips
- [x] Remove stroke
- [x] Change background color for – `chip_bg_selected_`

- - - - - -

### Context menu

- [x] Should be saved as component that should be reused
- [x] Text size - 16 sp
- [x] Checkbox default color – `icon_color_secondary_`

- - - - - -
 
<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><img src="https://github.com/user-attachments/assets/1d0852d8-84ad-4805-97ea-c4af183acc40"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Left / right padding - 12 dp
- [x] Padding between control or icon and text = 12 dp

- - - - - -
 
<details>
    <summary>Screenshot</summary>
    <table style="width:100%">
      <tr>
        <th></th>
      </tr>
      <tr>
        <td><video src="https://github.com/user-attachments/assets/477c9184-6eb9-472d-bf92-752fd7aa7d4d"  width="300" height="auto"/> </td>
      </tr>
    </table>
</details>  

- [x] Show second line while changing sorting order to avoid list jumps
